tensorrt_llm/llmapi/llm_args.py

```diff
@@ -1219,11 +1227,41 @@ def _setup(self):
         else:
             self.decoding_config = None
 
+    def _ensure_lora_config_consistency(self):
         if self.lora_config:
+            if self.max_lora_rank is not None:
+                logger.warning(
+                    "max_lora_rank is ignored when lora_config is provided.")
+            if self.max_loras != self.lora_config.max_loras:
+                logger.warning(
+                    "max_loras is ignored when lora_config is provided.")
+            if self.max_cpu_loras != self.lora_config.max_cpu_loras:
+                logger.warning(
+                    "max_cpu_loras is ignored when lora_config is provided.")
+
+            if len(self.lora_config.lora_dir) == 0:
+                # TODO [TRTLLM-5173]
+                logger.warning(
+                    "lora_dir is empty, so custom embedding or lm head will not be applied."
+                )
+
+        if self.enable_lora and self.lora_config is not None and self.backend == 'pytorch':
             logger.warning(
-                "Lora is an experimental feature and is probably subject to change before 1.0 release"
+                "enable_lora is ignored when lora_config is provided for pytorch backend."
             )
 
+        if self.lora_config is not None:
+            if len(self.lora_config.lora_dir) == 0 and len(
+                    self.lora_config.lora_target_modules) == 0:
+                logger.warning(
+                    "Both lora_dir and lora_target_modules are empty, so all LoRA modules will be expected. "
+                    "This will lead to serious memory consumption. Please provide either lora_dir or lora_target_modules."
```

Comment:
Should be better to say "Please provide either lora_dir or lora_target_modules if this behavior is not what you expect.`